miso-web/src/main/webapp/scripts/hot_library.js

```diff
@@ -37,40 +37,82 @@ HotTarget.library = (function() {
-        var indices = (Utils.array.maybeGetProperty(Utils.array.findFirstOrNull(function(family) {
-          return family.name == value && family.platformType == pt;
-        }, Constants.indexFamilies), 'indices') || []).filter(function(index) {
-          return index.position == n;
-        }).map(function(index) {
-          return index.label;
-        }).sort();
-        var data;
-        if (indices.length == 0 || n > 1) {
-          indices.unshift('No index');
-          data = 'No index';
-        } else {
-          data = '';
+      depends: ['indexFamilyName', 'templateAlias', 'boxPosition'],
+      update: function(lib, flat, flatProperty, value, setReadOnly, setOptions, setData) {
+        var indexFamily = null;
+        if (flatProperty === 'indexFamilyName') {
+          var pt = getPlatformType(flat.platformType);
+          indexFamily = Utils.array.findFirstOrNull(function(family) {
+            return family.name == flat.indexFamilyName && family.platformType == pt;
+          }, Constants.indexFamilies);
+          var indices = (Utils.array.maybeGetProperty(indexFamily, 'indices') || []).filter(function(index) {
+            return index.position == n;
+          }).map(function(index) {
+            return index.label;
+          }).sort();
+          var data;
+          if (indices.length == 0 || n > 1) {
+            indices.unshift('No index');
+            data = 'No index';
+          } else {
+            data = '';
+          }
+          setOptions({
+            'source': indices
+          });
+          setData(data);
         }
-        setOptions({
-          'source': indices
-        });
-        setData(data);
+        var readOnly = false;
+        if (flat.templateAlias && flat.boxPosition && indexFamily) {
+          var template = getTemplate(config, lib.parentSampleProjectId, flat.templateAlias);
+          var positionProp = n == 1 ? 'indexOneIds' : 'indexTwoIds';
+          if (template.indexFamilyId && template[positionProp] && template[positionProp][flat.boxPosition]) {
+            var index = Utils.array.getObjById(template[positionProp][flat.boxPosition], indexFamily.indices);
+            if (index) {
+              setData(index.label);
+              readOnly = true;
+            }
+          }
+        }
+        setReadOnly(readOnly);
       }
     };
   };
 
-  var makeDesignUpdate = function(id, nameProperty, defaultValue, source) {
-    return function(lib, flat, value, setReadOnly, setOptions, setCell) {
-      var design = Utils.array.findFirstOrNull(Utils.array.namePredicate(value), Constants.libraryDesigns);
-      if (design) {
-        setCell(Utils.array.maybeGetProperty(Utils.array.findFirstOrNull(Utils.array.idPredicate(design[id]), source), nameProperty)
-            || defaultValue);
+  var getTemplate = function(config, projectId, templateAlias) {
+    return Utils.array.findFirstOrNull(Utils.array.aliasPredicate(templateAlias), config.templatesByProjectId[projectId]);
```

Comment:
`config.templatesByProjectId` can be null (set on EditLibraryController line 689), causing this check to error.